drivers/spi/spi_context.h

```diff
@@ -183,9 +183,9 @@ static inline void _spi_context_cs_control(struct spi_context *ctx,
 {
 	if (ctx->config && ctx->config->cs && ctx->config->cs->gpio_dev) {
 		if (on) {
-			gpio_pin_write(ctx->config->cs->gpio_dev,
-				       ctx->config->cs->gpio_pin,
-				       spi_context_cs_active_value(ctx));
+			gpio_pin_set(ctx->config->cs->gpio_dev,
+				     ctx->config->cs->gpio_pin,
+				     spi_context_cs_active_value(ctx));
```

Comment:
We should use one of the two solutions:
- since GPIO driver can now work with logical levels the active level should be defined be the board DTS and the `SPI_CS_ACTIVE_HIGH` flag should only be used by the SPI module drivers.
- if we prefer to have consistent behavior and always honor `SPI_CS_ACTIVE_HIGH` flag we should call `gpio_pin_set_raw` to work with physical levels.

The current approach is inconsistent.

Reply:
@mnkp, this part of the change was  #21065, and it has been merged already.
I'm about to rebase the change and make this disappear, but I'll work on it as a separate PR.

Reply:
@mnkp, @tbursztyka what about adding a `gpio_flag` to `struct spi_cs_control` ?

Reply:
SPI_CS_ACTIVE_HIGH is not gpio emulated CS specific so it would be a bad idea to expose more gpio tweaks via the spi_cs_control structure I think.

Why not just using GPIO_DS_DFLT_LOW  / GPIO_DS_DFLT_HIGH depending on SPI_CS_ACTIVE_HIGH in spi_context.h ?  (I guess that's what @mnkp meant in the 2nd solution).

Note that, atm, none of the spi.h configuration bits are exposed in DTS and that's an issue that will need to be fixed. So in a near future SPI_CS_ACTIVE_HIGH will be accessible in DTS (whether the CS is gpio emulated or not).

Reply:
I had another look into how Linux DTS handles pins such as reset or CS and it seems like it always allows to configure pin active level via DTS. I.e. any driver that needs to use a GPIO pin always honors DTS configuration of the pin its using.

I propose that we do the same but in this case the approach taken in #21065 is invalid. The active level should be defined by `GPIO_ACTIVE_LOW`, `GPIO_ACTIVE_HIGH` flags in DTS and not `SPI_CS_ACTIVE_HIGH`.

Looking at the existing Zephyr spi drivers, only two (spi_cc13xx_cc26xx.c, spi_litespi.c) contain reference to `SPI_CS_ACTIVE_HIGH` and only to notify the user that the flag is not supported. It means that no existing driver is using `SPI_CS_ACTIVE_HIGH` to configure its CS pin. It would be easiest to simply deprecate `SPI_CS_ACTIVE_HIGH` and start using DTS configuration right now.

This also means that we would need to add `gpio_flag` to `struct spi_cs_control`.

Reply:
> It would be easiest to simply deprecate SPI_CS_ACTIVE_HIGH and start using DTS configuration right now.

Our drivers cannot be taken as master reference here. Plus, why removing a generic option (not tighten to gpio emulated cs), to let the possibility only existing for gpio emulated cs?

I mean, if the only flag to be used is about that cs active low/high (and which can be already inferred from SPI_CS_ACTIVE_HIGH) then no: I won't let the spi_cs_control to grow just for that.

Reply:
Adding `gpio_flag` to `struct spi_cs_control` means that we would support other DTS flags allowing user to configure pin as pull-up/pull-down, open-drain/open-source. As I mentioned, Linux drivers always honor GPIO DTS flags. That sounds like a reasonable approach, we should do the same.

However, the `GPIO_ACTIVE_LOW`, `GPIO_ACTIVE_HIGH` flags are indeed not generic. In case CS pin is controlled directly by SPI module it should be described in DTS using pinctrl bindings which don't allow to configure pin active level. In this case we would need `cs-active-high` (or similar) DTS property that would be used by the drivers that need it.

Reply:
> That sounds like a reasonable approach, we should do the same.

Just because linux is doing it is not a proper argument, imo. I am not against such support, but the problem is that no one has expressed this as a requirement. Everything has been working without this so far. So why adding features that haven't been necessary yet?

We've been there before, creating features in APIs that proved to be totally useless in the end. (yes, for instance, SPI_CS_ACTIVE_HIGH has yet to be used by some controller...)

So, just get spi_context.h to work again new gpio API, and let's see in the future if really someone will need such fine grain tweaks for cs gpio.

Reply:
@tbursztyka I knew you're going to disagree :-) OK, let's stick with the current solution, it's working. We can modify it later.